### PR TITLE
convert lists to arrays for tuple indexing

### PR DESCRIPTION
**Pull request by bqpd**
closes https://github.com/hoburg/d8/issues/37

@whoburg, ready for review

### CODE REVIEW

**Comment by bqpd** [comment]
todo: implement this or something similar as a test:

```python
from gpkit import *
from gpkit.constraints.relax import ConstantsRelaxed

x = VectorVariable(3, "x")
y = VectorVariable(3, "y")
ymax = VectorVariable(3, "ymax")

with SignomialsEnabled():
    m = Model(x.prod(), [x + y >= 1, y <= ymax])

m.substitutions.update({"ymax": [0.3, 0.5, 0.8]})
print m.localsolve().summary()
```

**Comment by mayork** [comment]
@whoburg it'd really help us do some much needed clean up in the D8 code if this gets merged

File: gpkit/keydict.py [hunk 1]
```diff
@@ -61,6 +62,9 @@ def update(self, *args, **kwargs):
                 #   KeyDict values are expected to be immutable (Numbers)
                 #   or to have a copy attribute.
                 v = v.copy()
+            if (isinstance(k, VarKey) and k.shape
```
Comment (whoburg) [inline gpkit/keydict.py]: move `isinstance` call to the end of the list of conditions, for speed?

File: gpkit/keydict.py [hunk 2]
```diff
@@ -61,6 +62,9 @@ def update(self, *args, **kwargs):
...
+                and not k.idx and not isinstance(v, np.ndarray)):
```
Comment (whoburg) [inline gpkit/keydict.py]: inline comment desperately needed.

**Comment by whoburg** [comment]
added @bqpd's suggested test.

**Comment by whoburg** [comment]
I set the review status to approved so that @bqpd can merge, but please add the inline comment first

**Comment by mayork** [comment]
@bqpd updates?

**Comment by bqpd** [comment]
@mayork, it's merged!

**Pull request merged**